### DISCUSSION TITLE
Nats still recommended?

### DISCUSSION THREAD

**Discussion by h0jeZvgoxFepBQ2C** [anycable/anycable#204] [Q&A] [answered]
I saw that anycable offers configuration options for NATS - but when I look at the adapter support, it shows NATS support is only legacy?

Is it still working to use nats or recommended? Why is it legacy?

**Comment by Envek** [comment]
At the current moment it is supported and recommended (and actually the simplest way to deploy AnyCable on fly.io for example), but in future versions of AnyCable there will be changes in pubsub architecture that will deprecate current adapters (and maybe there will be newer version of NATS adapter then). See this page for details: https://docs.anycable.io/edge/anycable-go/pubsub

As for now please use it without any hesitation!

**Reply by h0jeZvgoxFepBQ2C** [reply]
Could you maybe elaborate more what you mean with "and maybe ther will be a newer version")?

Since I wouldn't like to start our infrastructure with NATS, if it gets deprecated in the next major version of anycable, I would like to know how I should start :)

**Reply by palkan** [reply, marked as answer]
NATS is not gonna be deprecated. In the new architecture, there are two independent components which can use NATS: _broadcaster_ (responsible for receiving broadcast messages) and _pubsub_ (responsible for distributing broadcast messages within a cluster). We support NATS for in-cluster pub/sub in the upcoming versions, it's already there. For broadcasting, we will add NATS support, too. Right now, it's possible to use NATS pub/sub with the HTTP broadcasting in v1.4.

If you don't rely on [consistency features](https://docs.anycable.io/edge/anycable-go/broker), you can continue using the legacy NATS broadcaster. It's not deprecated, and we don't have plans on deprecating the current architecture. In our case, _legacy_ != _will be removed soon_. It's just a different architecture.

**Reply by h0jeZvgoxFepBQ2C** [reply]
Thank you very much @palkan for this detailed reply, this cleared it up pretty well! Thank you so much!! ❤️
